File: Java/ArmstrongNumber.java

// Github user name=  Tejasravi64
import java.util.Scanner;  
import java.lang.Math;  
public class ArmstrongNumber 
{  
static boolean isArmstrong(int n)   
{   
int temp, digits=0, last=0, sum=0;   
temp=n;    
while(temp>0)    
{   
temp = temp/10;   
digits++;   
}   
temp = n;   
while(temp>0)   
{       
last = temp % 10;     
sum +=  (Math.pow(last, digits));    
temp = temp/10;   
}  
if(n==sum)   
return true;      
else return false;   
}    
public static void main(String args[])     
{     
int num;   
Scanner sc= new Scanner(System.in);  
System.out.print("Enter the limit: ");   
num=sc.nextInt();  
System.out.println("Armstrong Number up to "+ num + " are: ");  
for(int i=0; i<=num; i++)   
if(isArmstrong(i)) 
System.out.print(i+ ", ");  
}   
}  

File: Armstrong_number.cpp

#include<iostream>
using namespace std;

int main(){
    int num,m,r=0,num1;
    cout<<"Enter the number to be checked: "<<endl;
    cin>>num;
    num1=num;
    while(num>0){
        m=num%10;
        r=(m*m*m)+r;
        num=num/10;
    }
    if(num1==r){
        cout<<"The entered number is armstrong number"<<endl;
    }
    else{
        cout<<"The entered number is not armstrong number"<<endl;
    }
    return 0;
}